### DISCUSSION TITLE
How can i use BrowserRouter basename in RouterProvider from v6.4?

### DISCUSSION THREAD

**Discussion by nihil-pro** [remix-run/react-router#9481] [Q&A] [answered]
Hi
Is there a way to set a basename to RouterProvider? 
I'am using react 17 now with such style
```javascript
   <BrowserRouter basename="/app/">
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <SnackbarProvider>
          <Routes>
	    <Route path='/' element={<Template />}>
		<Route index element={<HomePage />} />
                ...
            </Route>
        </SnackbarProvider>
      </ThemeProvider>
    </BrowserRouter>
```
How can i migrate to react 18 and react router 6.4?
Thanks

**Comment by brophdawg11** [comment, marked as answer]
You can provide a `basename` to `createBrowserRouter`:

```jsx
let router = createBrowserRouter(routes, { basename: "/base" })

function App() {
  return <RouterProvider router={router} />
}
```

**Reply by nihil-pro** [reply]
Thanks
This is not in the documentation [https://reactrouter.com/en/main/routers/router-provider](url)

**Reply by brophdawg11** [reply]
It's documented with `createBrowserRouter`: https://reactrouter.com/en/main/routers/create-browser-router#basename

**Reply by MiteshDarda** [reply]
What is routes here, it is giving an error saying that its not defined

**Reply by brophdawg11** [reply]
It's your route definitions https://reactrouter.com/en/main/routers/create-browser-router#routes

**Reply by dreygur** [reply]
Hi, I have used this approach in my project. And it's working fine on development. But doesn't load assets after build. Any thoughts?
